api/src/main/java/org/openmrs/validator/ImplementationIdValidator.java

```diff
@@ -40,17 +40,18 @@ public boolean supports(Class<?> clazz) {
 
 	@Override
 	public void validate(Object obj, Errors errors) throws APIException {
+
+		if (obj == null || !(obj instanceof ImplementationId)) {
+			throw new APIException("The parameter obj should not be null and must be of type" + ImplementationId.class);
+		}
+		
 		ImplementationId implId = (ImplementationId) obj;
 		char[] illegalChars = { '^', '|' };
-		if (implId == null) {
-			throw new APIException("ImplementationId.null", (Object[]) null);
-		} else {
-			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "ImplementationId.name.empty");
-			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "implementationId", "ImplementationId.implementationId.empty");
-			ValidationUtils.rejectIfEmptyOrWhitespace(errors, "passphrase", "ImplementationId.passphrase.empty");
-			if (implId.getImplementationId() != null && StringUtils.containsAny(implId.getImplementationId(), illegalChars)) {
-				errors.rejectValue("implementationId", "ImplementationId.implementationId.invalidCharacter");
-			}
+		ValidationUtils.rejectIfEmptyOrWhitespace(errors, "name", "ImplementationId.name.empty");
```

Comment:
@bejanut  thanks for working on this,did you make any changes in this and the following lines?

Reply:
No, I left everything as it was, I only refactored the code because I made sure that "implId" was not null and the previous check was irrelevant.